src/v/storage/disk_log_impl.cc

```diff
@@ -1545,8 +1545,11 @@ size_t disk_log_impl::max_segment_size() const {
 }
 
 uint64_t disk_log_impl::size_bytes_after_offset(model::offset o) const {
+    if (_segs.empty()) {
+        return 0;
+    }
     uint64_t size = 0;
-    for (size_t i = _segs.size() - 1; i-- > 0;) {
+    for (size_t i = _segs.size(); i-- > 0;) {
```

Comment:
This loop could be so much more simple with standard container instead of `ss::circular_buffer` in the `segment_set`.

There is a comment from Alex
```
    // NOTE: gcc has an ABI problem and cannot make std::deque noexcept
    // so we use a circular instead of a dequeue that *is* noexcept ctor
    // and allow us to truly have an empty container at ctor time
    // We loose reverse-iterators tho
```

maybe its time to revisit this?

Reply:
Otherwise the change looks good. Going to approve the PR.